test/unit/ConstantProductPair.t.sol

```diff
@@ -267,6 +279,46 @@ contract ConstantProductPairTest is BaseTest
         _constantProductPair.swap(-int256(INITIAL_MINT_AMOUNT + 1), false, address(this), bytes(""));
     }
 
+    function testSwap_ExactOut(uint256 aAmountOut) public
+    {
+        // arrange
+        vm.prank(address(_factory));
+        _constantProductPair.setCustomSwapFee(0);
+        uint256 lMinNewReservesOut = INITIAL_MINT_AMOUNT ** 2 / type(uint112).max + 1;
+        // this amount makes the new reserve of the input token stay within uint112 and not overflow
+        uint256 lMaxOutputAmt = INITIAL_MINT_AMOUNT - lMinNewReservesOut;
+        uint256 lAmountOut = bound(aAmountOut, 1, lMaxOutputAmt);
+        (uint256 lReserve0, uint256 lReserve1, ) = _constantProductPair.getReserves();
+        uint256 lAmountIn = _calculateInput(lReserve0, lReserve1, lAmountOut, _constantProductPair.swapFee());
```

Comment:
```suggestion
        // assume
        uint256 lMinNewReservesOut = INITIAL_MINT_AMOUNT ** 2 / type(uint112).max + 1;
        // this amount makes the new reserve of the input token stay within uint112 and not overflow
        uint256 lMaxOutputAmt = INITIAL_MINT_AMOUNT - lMinNewReservesOut;
        uint256 lAmountOut = bound(aAmountOut, 1, lMaxOutputAmt);
        
        // arrange
        vm.prank(address(_factory));
        _constantProductPair.setCustomSwapFee(0);
        (uint256 lReserve0, uint256 lReserve1, ) = _constantProductPair.getReserves();
        uint256 lAmountIn = _calculateInput(lReserve0, lReserve1, lAmountOut, _constantProductPair.swapFee());
```

Reply:
Make sure `vm.assume` or `bound` is always at top of test or even in its own section

Reply:
noted, will do so in the future. Fixed this as well as the corresponding test for `StablePair`. 

Will also do a refactor in the future to take care of all the cases where we use `bound` or `assume`.